### PR TITLE
build correct data source type

### PR DESCRIPTION
**Pull request by snopoke**
PR into https://github.com/dimagi/commcare-hq/pull/30285 (Safety info removed)

### CODE REVIEW

File: corehq/apps/userreports/reports/builder/forms.py [hunk 1]
```diff
@@ -868,7 +908,9 @@ def __init__(self, domain, registry_slug, source_type, source_id):
         registry_helper = DataRegistryHelper(self.domain, registry_slug=self.registry_slug)
         owning_domain = registry_helper.registry.domain
         prop_map = get_data_dict_props_by_case_type(owning_domain)
-        self.case_properties = sorted(set(prop_map[self.source_id]) | {'closed', 'closed_on'})
+        self.case_properties = sorted(
+            set(prop_map[self.source_id]) | {'closed', 'closed_on'} - {COMMCARE_PROJECT}
```
Comment (kaapstorm) [inline corehq/apps/userreports/reports/builder/forms.py:912]: `-` has higher precedence than `|`:
```python
>>> {'foo', 'bar', 'qux'} | {'foo', 'baz'} - {'qux'}
{'bar', 'baz', 'foo', 'qux'}
```
You probably want:
```python
>>> ({'foo', 'bar', 'qux'} | {'foo', 'baz'}) - {'qux'}
{'bar', 'baz', 'foo'}
```
i.e.:
```suggestion
            (set(prop_map[self.source_id]) | {'closed', 'closed_on'}) - {COMMCARE_PROJECT}
```

**Review by kaapstorm — COMMENTED** [review]
Looks good! The only thing blocking is the set operator precedence. Feel free to ignore my nit about function naming. :upside_down_face:

File: corehq/apps/userreports/reports/builder/forms.py [hunk 2]
```diff
@@ -1145,21 +1156,12 @@ def _get_data_source_configuration_kwargs(self):
...
-                # The uuid gets truncated, so it's not really universally unique.
-                table_id=clean_table_name(self.domain, str(uuid.uuid4().hex)),
-                **self._get_data_source_configuration_kwargs()
-            )
+    def _build_data_source(self):
+        data_source_config = self.ds_builder.construct_data_source(
+            # The uuid gets truncated, so it's not really universally unique.
+            table_id=clean_table_name(self.domain, str(uuid.uuid4().hex)),
+            **self._get_data_source_configuration_kwargs()
+        )
```
Comment (stephherbers) [inline corehq/apps/userreports/reports/builder/forms.py:1183]: this is much much cleaner-- I like it

File: corehq/apps/userreports/reports/filters/factory.py [hunk 3]
```diff
@@ -206,7 +206,8 @@ class FilterChoiceProviderFactory(object):
         DATA_SOURCE_COLUMN: DataSourceColumnChoiceProvider,
         LOCATION: LocationChoiceProvider,
         USER: UserChoiceProvider,
-        OWNER: OwnerChoiceProvider
+        OWNER: OwnerChoiceProvider,
+        COMMCARE_PROJECT: DomainChoiceProvider
```
Comment (stephherbers) [inline corehq/apps/userreports/reports/filters/factory.py:210]: ah thank you this was a big missing piece of oversight

File: corehq/apps/userreports/reports/builder/forms.py (hunk 1 above)
Comment (snopoke) [reply corehq/apps/userreports/reports/builder/forms.py:912]: Thanks, this isn't actually needed at all
```suggestion
            set(prop_map[self.source_id]) | {'closed', 'closed_on'}
```

**Comment by snopoke** [comment]
@stephherbers I'll leave you to address any feedback on this

**Review by kaapstorm — APPROVED** [review]

**Comment by stephherbers** [comment]
sounds good thanks @snopoke

**Pull request merged**